tests/test_model.py

```diff
@@ -163,6 +164,10 @@ def test_write_data_catalog(tmpdir):
     model1.data_catalog.get_source(sources[0]).mark_as_used()
     model1.write_data_catalog(append=True)
     assert list(DataCatalog(data_lib_fn).sources.keys()) == sources[:2]
+    # test writing table of datacatalog as csv
+    model.write_data_catalog(used_only=False, save_csv=True)
+    assert isfile(join(model.root, "hydromt_data.csv"))
+    assert not pd.read_csv(join(model.root, "hydromt_data.csv")).empty
```

Comment:
Can you make this check a bit more precise? E.g. test if `sources[0]` is on line 1 or that the number of lines matches the number of sources.